pkg/plugins/gateway/util.go

```diff
@@ -86,10 +86,14 @@ func validateRoutingStrategy(routingStrategy string) bool {
 // getRequestMessage returns input request message field which has user prompt
 func getRequestMessage(jsonMap map[string]interface{}) (string, *extProcPb.ProcessingResponse) {
 	messages, ok := jsonMap["messages"]
```

Comment:
I feel such change brings the compatibility issues. We can not make assumption whether users use /v1/completion or /v1/chat/completion api. 

for example, only chat api has `messages` and completion only have `prompt`. We should rely on backend pods to do such validation. 

https://github.com/vllm-project/aibrix/issues/732 Let's use this issue to drive related work

Reply:
Right now, if input request does not have "messages" field then routing-strategy based requests will fail. I agree this is not the best solution but good work around patch.

Reply:
>  if input request does not have "messages" field then routing-strategy based requests will fail

you mean the dependencies to parse the `model` body? If that's case, for short term workaround, we need to retrieve `model` from different body structure. `body.model` is kind of common for these two api but no guarantee for rest of apis. Let's document these design pitfalls..